modules/core/src/main/scala/pillars/codec.scala

```diff
@@ -36,14 +36,20 @@ object codec:
     given Decoder[FiniteDuration] = Decoder.decodeDuration.map(_.toScala)
     given Encoder[FiniteDuration] = Encoder.encodeDuration.contramap(_.toJava)
 
-    given Decoder[Duration] = Decoder.decodeDuration.map(_.toScala)
-    given Encoder[Duration] = Encoder.instance[Duration]:
-        case Duration.Inf       => "infinity".asJson
-        case Duration.MinusInf  => "-infinity".asJson
-        case Duration.Zero      => "0".asJson
-        case Duration.Undefined => "undefined".asJson
-        case d: FiniteDuration  => d.asJson
-        case other              => other.toString.asJson
+    given Decoder[Duration] = Decoder.decodeString.map {
+        case "infinity"  => Duration.Inf
+        case "-infinity" => Duration.MinusInf
+        case "0"         => Duration.Zero
+        case "undefined" => Duration.Undefined
+    } orElse Decoder.decodeDuration.map(_.toScala)
```

Comment:
Thanks for this change. But, would it be possible to use the braceless syntax? Also, I think infix notation should be avoided.

```suggestion
    given Decoder[Duration] = 
        Decoder.decodeString.map:
            case "infinity"  => Duration.Inf
            case "-infinity" => Duration.MinusInf
            case "0"         => Duration.Zero
            case "undefined" => Duration.Undefined
        .orElse(Decoder.decodeDuration.map(_.toScala))
```

Reply:
Changed code syntax to have clearer decoders, tell me what you think ! Also followed your syntax rules

Reply:
Hmmm found more edge cases, fixing them